src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
+
+    if (requestSecret && requestSecret === secret) {
+      return true;
+    } else {
+      throw new UnauthorizedException('Invalid feedback secret');
+    }
+  }
```

Comment:
PullProofAI

The implementation of the `FeedbackSecretGuard` class for validating a feedback secret header.

```diff
+ Suggested fix: No issues found in the provided code snippet.
```

The `FeedbackSecretGuard` class seems well-implemented for its intended purpose of guarding access based on a feedback secret. The code checks if the request contains a specific header with a secret value and throws an exception if the secret is invalid. This guard can help ensure that only authorized requests with the correct feedback secret are allowed to pass through, enhancing security in the application.


<!-- This is an auto-generated comment by PullProofAI -->